lib/rules/no-duplicate-selectors/__tests__/index.js

```diff
@@ -84,6 +84,12 @@ testRule(rule, {
     message: messages.rejected("a b"),
     line: 1,
     column: 12,
+  }, {
+    code: "a { & {} }",
+    description: "duplicate caused by &-parent selector",
+    message: messages.rejected("a"),
```

Comment:
It will be `a {} a {}` or `a {}`. We can't predict what preprocessor will do. Example http://www.sassmeister.com/gist/8ad7e0270cf5817804625228e74f9c6b